examples/js/loaders/GLTFLoader.js

```diff
@@ -96,7 +96,7 @@ THREE.GLTFLoader = ( function () {
 
 		},
 
-		parse: function ( data, path, onLoad, onError ) {
```

Comment:
`parse()` is part of the documented public API of this loader (https://threejs.org/docs/#examples/loaders/GLTFLoader.parse), so to prevent confusion on the next release could you (1) update the docs, and (2) log a warning if only `onProgress` is passed to the method?

```js
if ( onProgress && onError === undefined ) {

  console.warn( 'THREE.GLTFLoader: .parse() parameters have changed, callback order is now "onLoad, onProgress, onError".' );

  onError = onProgress;
  onProgress = undefined;

}
```

Reply:
And let's add a note to migration guide after this PR is merged. https://github.com/mrdoob/three.js/wiki/Migration-Guide

Reply:
About

if ( onProgress && onError === undefined ) {

    console.warn( 'THREE.GLTFLoader: .parse() parameters have changed, callback order is now "onLoad, onProgress, onError".' );

    onError = onProgress;
    onProgress = undefined;

}

I think only warning may be good enough because some users may want to set only `onLoad` and `onProgress`. `onError` is optional.

Reply:
yes, this will fix possible problems for someone who is switched from an old to the new version, but this can make problems in the future. I don't know what bugs are more important to prevent right now?

Reply:
I think user should be able to do `loader.parse( data, path, onLoad, onProgress, undefined )` because it's valid.

Reply:
in that case, when you don't set onError we will make a mess using progress function in error case. So user must to explicitly set onError to null. We can:
1. remove this swap, but users who don't check MigrationGuide and console may have problems
2. move onProgress as last (new) parameter, so everything will work fine with old code. Problem is that we will change usual callback order convention.

I will change whatever you think is better solution.

Reply:
I prefer 1. Any thoughts? @donmccurdy @mrdoob

Reply:
BTW, `onXXX` callback functions should be `undefined` or `function`. I don't think passing `null` is a good idea.